website/docs/docs/build/unit-tests.md

```diff
@@ -319,6 +322,36 @@ Exit codes differ from data test success and failure outputs because they don't
 Learn about [exit codes](/reference/exit-codes) for more information.
 
 
+### Common Pitfalls
+> - **Missing Fixtures for Referenced Models**: When creating a unit test, all referenced models must be declared as mock inputs. Missing any referenced model, even if it isn't directly involved in the specific logic being tested, will lead to compilation errors such as "node not found."
+> - **Not Satisfying `WHERE` or `JOIN` Logic**: Ensure that the mock data meets all conditions in the model, such as `WHERE` clauses or `JOIN` requirements. If these conditions are not met, the unit test will either return null rows or fail to execute properly. This often involves adding rows for auxiliary data tables, like locations or transactions, to satisfy joins and filters.
+
+### How Unit Tests Compile
+> During a unit test, dbt creates Common Table Expressions (CTEs) for all dependencies of the model using the mock input data you provide. These CTEs replace the actual references (`ref()`) in the model and allow dbt to run your SQL logic against the mock data.
+>
+> For example, when you provide a reference such as `ref('stg_transactions')`, dbt creates a CTE named `__dbt__cte__stg_transactions` that contains the mocked data. The entire compiled SQL might look something like this:
+> ```sql
+> with
+>  __dbt__cte__stg_transactions as (
+>   -- fixture for stg_transactions
+>   -- contains unions to create "test inputs" corresponding to all rows
+> ),
+>  __dbt__cte__stg_locations as (
+>   -- fixture for stg_locations
+>   -- contains select statement that "mocks" stg_locations
+> ),
+> applied_donations as (
+>  select
+>    transaction_id,
+>    sum(cash_value) as donated_cash
+>  from __dbt__cte__stg_donations
+>  group by transaction_id
+> )
+> select * from __dbt__cte__stg_transactions;
+> ```
+> Understanding this process will help ensure you configure your unit tests correctly and avoid common issues.
+
+
```

Comment:
This is interesting implementation detail that helps in troubleshooting when there is some kind of error. In general, this kind of detail doesn't feel like it should go in product docs though because the maintainers are free to change the details at will. Maybe we move it to Markdown docs within the dbt-core or dbt-adapters repo instead?